File: server/Models/Req/RegisterReq.cs
using System.ComponentModel.DataAnnotations;

namespace ConnectoDb.Server.Models.Req;

public class RegisterReq
{
    [Required, MinLength(3), MaxLength(256)]
    public string Username { get; set; } = null!;

    [Required, MinLength(8), MaxLength(256)]
    public string Password { get; set; } = null!;

    [MaxLength(256)]
    public string? FirstName { get; set; }

    [MaxLength(256)]
    public string? LastName { get; set; }
}

File: clients/java/src/main/java/com/connecto/client/models/RegisterReq.java
package com.connecto.client.models;

import com.google.gson.annotations.SerializedName;

public class RegisterReq {
    public String username;
    public String password;
    @SerializedName("firstName")
    public String firstName;
    @SerializedName("lastName")
    public String lastName;

    public RegisterReq(String username, String password) {
        this.username = username;
        this.password = password;
    }
}
